### DISCUSSION TITLE
Is it possible to ignore a script in a specific package?

### DISCUSSION THREAD

**Discussion by mythmon** [vercel/turborepo#9528] [Help] [answered]
### Summary

I'm working on migrating an existing large repository to use yarn workspaces and turborepo. There are a few NPM scripts that exist in the `package.json`s that don't work without additional set up. For example, in one package I'm trying to integrate, the "test" npm only works inside a specific Docker container, in another only with certain environment variables set up.

Of course the long term solution is to make these NPM scripts more "standalone" so that they work without the extra setup. I'm looking for a quicker solution so we can get a broad integration done before optimizing it.

Is it possible tell turbo to ignore specific scripts? We have "test" as a Turborepo task, but I'd like to ignore "package1#test" while still running "package2#test". I am hoping for something that is in the turbo config so that I can set it globally instead of having to filter ever turbo call.

**Comment by anthonyshew** [comment]
There isn't a way to do exactly what you're describing today. However, it sounds like a workaround might be to get creative with script naming. For instance:
```
// turbo.json
{
  "tasks": {
    "test": { // configuration }
    "test-but-only-in-a-docker-container": { // configuration }
  }
}
```

Then, you'd need to rename the script in `package.json` for that package. It would then be it's own separate task that you'd could give special treatment.

Note: You could also choose to use a [Package Configuration](https://turbo.build/repo/docs/reference/package-configurations) if this is a one-off for a specific package, to get that task definition localized.

Let me know if that helps. I could not taking account an important piece of your puzzle.

**Reply by mythmon** [reply, marked as answer]
Package configuration sounds useful, but in this case I don't want Turbo to ever run a particular task. For now I've gone with "test-not-turbo". Not very elegant, but it works.

**Comment by nerdyman** [comment]
I'm running into a similar issue and am using the `--filter="!my-app"` flag on the turbo commands as a workaround but I have to remember to put it on each command which is cumbersome.

The task I need to filter on is a dependency of other tasks so I also have to manually apply the filter to those commands too. I don't really want to have to change the underlying scripts in my packages since they use bog standard `test` and `build` commands that work out of the box with platforms like Vercel.

I'm doing this atm:

```sh
turbo run build --filter="!my-app"
turbo run test --filter="!my-app" # test depends on build, if this filter isn't set build will run for `my-app`
```

Would it be feasible to have a `filter` property set at the task level in the config so it doesn't need to be manually applied for each command/dependency?

E.g.

```jsonc
"build": {
  "filter": ["!my-app"], <-- Any tasks dependent on `build` will also skip `my-app`
  "dependsOn": ["generate", "^build"],
  "outputs": ["dist/**", ".next/**"]
},
"test": {
  "dependsOn": ["build", "^test"],
  "outputs": ["coverage/**"]
},
```
